bitwarden_license/src/Commercial.Core/AdminConsole/Providers/RemoveOrganizationFromProviderCommand.cs

```diff
@@ -75,13 +84,44 @@ public class RemoveOrganizationFromProviderCommand : IRemoveOrganizationFromProv
 
         await _stripeAdapter.CustomerUpdateAsync(organization.GatewayCustomerId, customerUpdateOptions);
 
-        var subscriptionUpdateOptions = new SubscriptionUpdateOptions
+        var isConsolidatedBillingEnabled = _featureService.IsEnabled(FeatureFlagKeys.EnableConsolidatedBilling);
+
+        if (isConsolidatedBillingEnabled && provider.Status == ProviderStatusType.Billable)
         {
-            CollectionMethod = "send_invoice",
-            DaysUntilDue = 30
-        };
+            var plan = StaticStore.GetPlan(organization.PlanType).PasswordManager;
+            var subscription = await _stripeAdapter.SubscriptionCreateAsync(new SubscriptionCreateOptions
+            {
+                Customer = organization.GatewayCustomerId,
+                CollectionMethod = "send_invoice",
+                DaysUntilDue = 30,
+                AutomaticTax = new SubscriptionAutomaticTaxOptions { Enabled = true },
+                Metadata = new Dictionary<string, string>
+                {
+                    { "organizationId", organization.Id.ToString() }
+                },
+                OffSession = true,
+                ProrationBehavior = StripeConstants.ProrationBehavior.CreateProrations,
+                Items = new List<SubscriptionItemOptions>
+                {
+                    new SubscriptionItemOptions { Price = plan.StripeSeatPlanId, Quantity = organization.Seats }
+                }
+            });
+            organization.GatewaySubscriptionId = subscription.Id;
+        }
+        else
+        {
+            var subscriptionUpdateOptions = new SubscriptionUpdateOptions
+            {
+                CollectionMethod = "send_invoice",
+                DaysUntilDue = 30
+            };
+            await _stripeAdapter.SubscriptionUpdateAsync(organization.GatewaySubscriptionId, subscriptionUpdateOptions);
+        }
+
+        await _organizationRepository.ReplaceAsync(organization);
 
-        await _stripeAdapter.SubscriptionUpdateAsync(organization.GatewaySubscriptionId, subscriptionUpdateOptions);
+        await _scaleSeatsCommand.ScalePasswordManagerSeats(provider, organization.PlanType,
```

Comment:
❌ Scaling Password Manager seats for a Provider is only for Consolidated Billing and should be in the block with the feature flag.

Reply:
resolved